src/simularium/WebsocketClient.ts

```diff
@@ -150,18 +150,15 @@ export class WebsocketClient {
             // Handle binary message
             const floatView = new Float32Array(event.data);
             const binaryMsgType = floatView[0];
-            for (let msgType in this.binaryMessageHandlers) {
-                if (binaryMsgType == Number(msgType)) {
-                    this.binaryMessageHandlers[msgType](event);
-                    // TODO: is it ok if we have multiple handlers for the same
-                    // message type? should we return here? for now, we will
-                    return;
-                }
+            if (binaryMsgType in this.binaryMessageHandlers) {
```

Comment:
I like this change and looking at it, it looks like a great candidate for some kind of unit test.

Reply:
ok so I just wrote out some unit tests for this, only to realize that I'd need to change `onMessage` to be `public` instead of `protected` for the tests to run... how do we feel about that?

Reply:
Gah I stand corrected, the typechecker doesn't like that I'm trying to call `protected onMessage()` from my test, is there a hacky way around that or should I change `onMessage` to public or do you potentially have a different suggestion? @toloudis

Reply:
One way is to do this: in your test make a class that derives from the main class, and add a public function that calls the protected one.  It's kind of a nasty workaround but I bet it would do the trick.

```
class TestMyClass extends MyClass {
  public exposeProtectedFuncForTesting() { return protectedFunc(); }
}
```

Reply:
I guess I'm not strictly opposed to making onMessage public, either.  Another way is to refactor the code to make the onMessage in its own little class where it makes more sense to be public.